### PR TITLE
feat(ingestion/snowflake): use user email-id in urn generation for top users stat

### PR DESCRIPTION
**Pull request by mohdsiddique**

### CODE REVIEW

File: metadata-ingestion/src/datahub/ingestion/source/snowflake/snowflake_config.py [hunk 1]
```diff
@@ -120,6 +120,11 @@ class SnowflakeV2Config(
         "upstreams_deny_pattern", "temporary_tables_pattern"
     )
 
+    email_as_user_identifier: bool = Field(
+        default=False,
```
Comment (asikowitz) [inline metadata-ingestion/src/datahub/ingestion/source/snowflake/snowflake_config.py:124]: Let's make this default True and add a breaking change in `docs/how/updating-datahub.md`

File: metadata-ingestion/src/datahub/ingestion/source/snowflake/snowflake_usage_v2.py [hunk 2]
```diff
@@ -357,7 +366,11 @@ def _get_operation_aspect_work_unit(
             )
             reported_time: int = int(time.time() * 1000)
             last_updated_timestamp: int = int(start_time.timestamp() * 1000)
-            user_urn = make_user_urn(self.get_user_identifier(user_name, user_email))
+            user_urn = make_user_urn(
+                self.get_user_identifier(
+                    user_name, user_email, self.config.email_as_user_identifier
```
Comment (asikowitz) [inline metadata-ingestion/src/datahub/ingestion/source/snowflake/snowflake_usage_v2.py:371]: Should we add logic to generate an email from `self.config.email_domain` if it's set, like we do in `_map_user_counts`? Perhaps it'd be best to put that logic in `get_user_identifier` instead

File: metadata-ingestion/src/datahub/ingestion/source/snowflake/snowflake_usage_v2.py [hunk 3]
```diff
@@ -247,7 +249,9 @@ def _map_top_sql_queries(self, top_sql_queries: Dict) -> List[str]:
             ]
         )
 
-    def _map_user_counts(self, user_counts: Dict) -> List[DatasetUserUsageCounts]:
+    def _map_user_counts(
+        self, user_counts: Dict, email_as_user_identifier: bool
```
[the remark this answers is not in the document]
Comment (mohdsiddique) [reply metadata-ingestion/src/datahub/ingestion/source/snowflake/snowflake_usage_v2.py:253]: done

File: metadata-ingestion/src/datahub/ingestion/source/snowflake/snowflake_usage_v2.py (hunk 2 above)
Comment (mohdsiddique) [reply metadata-ingestion/src/datahub/ingestion/source/snowflake/snowflake_usage_v2.py:371]: let's keep it as is. The current method uses the user_email to determine whether to skip the current metadata (i.e continue in for loop)

**Review by asikowitz — APPROVED** [review]
LGTM!

**Pull request merged**